### ISSUE TITLE
InvalidGroup.Duplicate security group

### ISSUE THREAD

**Issue by ethem-kinginthenorth** [dask/dask-cloudprovider#198]
**What happened**:
I am getting `botocore.exceptions.ClientError: An error occurred (InvalidGroup.Duplicate) when calling the CreateSecurityGroup operation: The security group 'dask-default_12348' already exists for VPC 'vpc-0e0'` error when running the code snippet below:

```
cluster = EC2Cluster(ami="ami-02043f4809ea36ba3",  
                             docker_image="rapidsai/rapidsai:cuda10.2-runtime-ubuntu18.04",
                             region='us-east-1',
                             instance_type="g4dn.xlarge",
                             worker_class="dask_cuda.CUDAWorker", 
                             bootstrap=False,
                             n_workers=2,
                             filesystem_size=120)
```
![ee](https://user-images.githubusercontent.com/44275619/100158755-b0adcc80-2e7a-11eb-9510-bdfe36ec3747.PNG)

**Environment**:

- Dask version: 2.30.0
- Python version: 3.7.8
- Operating System: linux
- Install method (conda, pip, source): pip

**Comment by quasiben** [comment]
@jacobtomlinson, is this because dask-cloudprovider didn't clean up or didn't have a chance to clean up properly (manual intervention) ?

**Comment by jacobtomlinson** [comment]
Sorry for the delay in getting back to this.

The thing that is confusing me here is where the `dask-default_12348` name is coming from. The default is `dask-default` as defined in this constant.

https://github.com/dask/dask-cloudprovider/blob/6cfb35faeb2fd4e06329375f4c930a0963fe5123/dask_cloudprovider/aws/helper.py#L4

The behaviour here is that if you do not pass a list of security group IDs then we try to create one for you. When creating the group we first check to see if there is an existing one with the default name, and if not we create it. This happens for every VM, so in theory when using for the first time the scheduler VM creates the group and then all subsequent VMs make use of it.

A workaround would be to manually specify an existing security group with appropriate permissions (ports `8786` and `8787` to wherever you are calling from and all traffic within the group).

```python
cluster = EC2Cluster(ami="ami-02043f4809ea36ba3",  
                     docker_image="rapidsai/rapidsai:cuda10.2-runtime-ubuntu18.04",
                     region='us-east-1',
                     instance_type="g4dn.xlarge",
                     worker_class="dask_cuda.CUDAWorker", 
                     security_groups=["sg-aaabbbccc111222333"],  # Your security group ID
                     bootstrap=False,
                     n_workers=2,
                     filesystem_size=120)
```

Perhaps it would be sensible to move this logic up to the cluster manager, so this runs once when the cluster object starts. But regardless I am unable to reproduce this.

I did come across another issue with launching `g4dn` instances where they take longer than other instance types to be listed once they have been created. I've raised #207 to resolve this.

**Comment by ethem-kinginthenorth** [comment]
I tried creating a cluster. Then in my second attempt I got this error. So to pass this I started changing the default name. What is more interesting is, I have used an environment that my collogue created and with that environment I didn't get that error. So now I am thinking that there should ne an AWS parameter causing this issue but my investigation didn't end up anything. 

That said, if you cannot reproduce this, I can close the issue assuming that AWS setting is the root cause of the issue.
